integration/core/test_api.py

```diff
@@ -399,7 +399,7 @@ def test_upgrade_links(client):
     resp = response.json()
     upgradeLinks = resp['upgradeVersionLinks']
     assert upgradeLinks is not None
-    assert len(upgradeLinks) == 10
+    assert len(upgradeLinks) == 1
```

Comment:
Here is the curl request and response for this (before your changes in this PR):
```
curl http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:1?projectId=1a5 | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1818  100  1818    0     0  63100      0 --:--:-- --:--:-- --:--:-- 64928
{
  "actions": {},
  "bindings": null,
  "files": {
    "README.md": "# Test Upgrade Links\n\n\n### Info:\n\n This template has more than 9 versions.",
    "docker-compose.yml": "test-upgrade-links:\n  image: busybox\n  tty: true\n  stdin_open: true\n",
    "rancher-compose.yml": ".catalog:\n  name: \"Test Upgrade Links\"\n  version: \"1.0.1\"\n  description: \"More than 9 versions\"\n  minimum_rancher_version: v0.46.0\n  upgrade_from:\n"
  },
  "id": "test:test-upgrade-links:1",
  "labels": {},
  "links": {
    "icon": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links?image&projectId=1a5",
    "readme": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:1?readme",
    "self": "http://localhost:8088/v1-catalog/templateversions/test:test-upgrade-links:1",
    "template": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links"
  },
  "maximumRancherVersion": "",
  "minimumRancherVersion": "v0.46.0",
  "questions": null,
  "templateId": "test:test-upgrade-links",
  "type": "templateVersion",
  "upgradeFrom": "",
  "upgradeVersionLinks": {
    "1.0.11": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:11",
    "1.0.12": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:12",
    "1.0.2": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:2",
    "1.0.3": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:3",
    "1.0.4": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:4",
    "1.0.5": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:5",
    "1.0.6": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:6",
    "1.0.7": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:7",
    "1.0.8": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:8",
    "1.0.9": "http://localhost:8088/v1-catalog/templates/test:test-upgrade-links:9"
  },
  "version": "1.0.1"
}
```
upgradeVersionLinks represents what this version (1.0.1) can upgrade to. The vast majorities of those versions do not have an upgrade from:
```
test-upgrade-links$ grep -Ir  upgrade_from .
./4/rancher-compose.yml:  upgrade_from:
./2/rancher-compose.yml:  upgrade_from:
./6/rancher-compose.yml:  upgrade_from:
./0/rancher-compose.yml:  upgrade_from:
./1/rancher-compose.yml:  upgrade_from:
./12/rancher-compose.yml:  upgrade_from:
./3/rancher-compose.yml:  upgrade_from:
./8/rancher-compose.yml:  upgrade_from:
./7/rancher-compose.yml:  upgrade_from:
./11/rancher-compose.yml:  upgrade_from: ">1.0.0 <1.0.10"
./10/rancher-compose.yml:  upgrade_from: ">=1.0.5"
./9/rancher-compose.yml:  upgrade_from:
./5/rancher-compose.yml:  upgrade_from:
```
so, they all should be valid upgrades except for the version corresponding to `../10/rancher-compose.ym./10/rancher-compose.yml`, right?